torch/csrc/jit/backends/backend_preprocess.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <torch/csrc/jit/backends/backend_detail.h>
+namespace torch {
+namespace jit {
+class preprocess_register {
+  std::string backend_name_;
+
+ public:
+  preprocess_register(
```

Comment:
We don't need a class for this, right?
I think for backend the reason is that it was needed for the torch binding mechanism of torch::class_<>()

Technically we just need to expose detail::registerBackendPreprocessFunction either directly, or through some other namespace.

My suggestion is to just expose it through a a function that calls detail::registerBackendPreprocessFunction, unless there's an advantage to the class?

In terms of the name I'd make it very clear like registerBackendPreprocessing or something like that.

The same for the file name; maybe be should at least name the file more like what we want in the future: backend_compiler_registry.h or something like that.

Reply:
Good point to make the name specific! I have changed it to backend_preprocess_register. 

The reason to make it a class is to have it stay in the global space. If I just call the function, registerBackendPreprocessing, in the global space, there is compiler error. The compiler expects it to be a function declaration. I was stuck in compilation error till I found some example of wrapping it to a class. In such a case, the class object "reg" would stay in the binary. Not sure if there's a better solution on this.